js_modules/dagit/packages/core/src/workers/runLogs/runLogs.tsx

```diff
@@ -0,0 +1,51 @@
+import {ApolloClient} from '@apollo/client';
+
+import {PIPELINE_RUN_LOGS_SUBSCRIPTION} from '../../runs/PipelineRunLogsSubscription';
+import {PipelineRunLogsSubscription} from '../../runs/types/PipelineRunLogsSubscription';
+
+export type Message = INITIALIZE;
+type INITIALIZE = {
+  type: 'INITIALIZE';
+  runId: string;
+  postMessage: (data: any) => void;
+  getApolloClient: () => ApolloClient<any>;
+  staticPathRoot: string;
+  rootServerURI: string;
+};
+
+export function onMainThreadMessage(data: Message) {
+  switch (data.type) {
+    case 'INITIALIZE':
+      initialize(data); // subscribes to the data using data argument
+      break;
+  }
+}
+
+const initialPostMessage = (_data: any): void => {
+  throw new Error('Worker not initialized');
+};
+let postMessage = initialPostMessage;
```

Comment:
Does this need to be tracked at the module level? I only see it referenced in `initialize`.